lib/matplotlib/axes/_subplots.py

```diff
@@ -117,22 +117,44 @@ def label_outer(self):
         """
         Only show "outer" labels and tick labels.
 
-        x-labels are only kept for subplots on the last row; y-labels only for
-        subplots on the first column.
+        x-labels are only kept for subplots on the last row (or first row, if
+        labels are on the top side); y-labels only for subplots on the first
+        column (or last column, if labels are on the right side).
         """
+        self._label_outer_xaxis()
+        self._label_outer_yaxis()
+
+    def _label_outer_xaxis(self):
+        ss = self.get_subplotspec()
+        label_position = self.xaxis.get_label_position()
+        if not ss.is_first_row():  # Remove top label/ticklabels/offsettext.
+            if label_position == "top":
+                self.set_xlabel("")
+            self.xaxis.set_tick_params(which="both", labeltop=False)
+            if self.xaxis.offsetText.get_position()[1] == 1:
+                self.xaxis.offsetText.set_visible(False)
```

Comment:
This part not tested?

Reply:
This is because moving offset texts is somewhat broken anyways (https://github.com/matplotlib/matplotlib/issues/19471).